apache2/waf_logging/waf_log_util.cc

```diff
@@ -167,12 +220,34 @@ void free_json(char* str) {
     std::free(str);
 }
 
-void init_appgw_rules_id_hash() {
-    std::ifstream ifs("/etc/nginx/modsec/disableableRules.conf");
+static rule_ids_vector read_disableable_rule_ids(const std::string& filename) {
+    rule_ids_vector rule_ids;
+    std::ifstream ifs(filename);
     if(ifs) {
         std::copy(std::istream_iterator<int>(ifs),
                 std::istream_iterator<int>(),
-                std::back_inserter(default_disableable_rule_ids));
-        std::sort(default_disableable_rule_ids.begin(), default_disableable_rule_ids.end());
+                std::back_inserter(rule_ids));
+        std::sort(std::begin(rule_ids), std::end(rule_ids));
+    }
+
+    return rule_ids;
+}
+
+void init_appgw_rules_id_hash() {
+    std::map<rule_set_info, std::string> pathsMap {
+        {OWASP_2_2_9, "/etc/nginx/modsec/OWASP/2.2.9/disableableRules.conf"},
+        {OWASP_3_0, "/etc/nginx/modsec/OWASP/3.0/disableableRules.conf"},
+        {OWASP_3_1, "/etc/nginx/modsec/OWASP/3.1/disableableRules.conf"},
+    };
+    for (const auto& item : pathsMap) {
+        auto vec = read_disableable_rule_ids(item.second);
+        if (!vec.empty()) {
```

Comment:
Throw an error if empty?

Reply:
It is not an error. It is legal and possible that we have not all of these files.
Imagine that we have 3 WAF policies for a gateway, two of them use 3.0 and one uses 2.2.9. In that case the tenant will only write files for 2.2.9 and 3.0 because writing a file for 3.1 would be redundant - noone is going to use it.

Here we just make sure that we only add existing lists of rule IDs to the map.

A better option here would be to return a `std::optional<rule_ids_vector>` which is only engaged when the file is present. However, our current version of compiler only supports `std::optional` through the `std::experimental` namespace and requires `-std=c++14` flag. I did not want to bother with that for reflecting a seemingly minor thing here yet, but if you think that conveys the intended logic here better I could definitely do that.

Reply:
Sounds good, was under the impression that all versions would be present at all times.